src/main/java/com/eprosima/fastcdr/idl/templates/TypeObjectSource.stg

```diff
@@ -59,23 +59,23 @@ $definitions; separator="\n"$
 
 
 try_create(ctx, object, param) ::= <<
-if ($param$ == "$if(object.hasScope)$$object.scope$::$endif$$object.name$") return $if(object.hasScope)$$object.scope$::$endif$Get$object.name$Identifier(complete);
+if ($param$ == $if(ctx.GenerateTypesROS2)$"$object.ROS2Scopedname$"$else$"$object.scopedname$"$endif$) return $if(object.hasScope)$$object.scope$::$endif$Get$object.name$Identifier(complete);
```

Comment:
I am not sure if  `object.scopedname` behavior is the same as the previous one which was:

```
if (object.hasScope)
{
    object.scope
}
object.name
```

Reply:
After checking with the first and second commit, I think that `object.scopedname` really has the desired behavior. Approving nevertheless as it is working.